app/src/main/java/nu/yona/app/ui/message/MessageStickyRecyclerAdapter.java

```diff
@@ -64,63 +71,137 @@ public MessageItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
+				currentMessageItemHolder.txtTitleMsg.setText(currentYonaMessage.getNotificationMessageEnum().getUserMessage());
 			}
-			if (yonaObject.getNotificationMessageEnum() == NotificationMessageEnum.SYSTEM_MESSAGE)
-			{
-				holder.img_avtar.setVisibility(View.GONE);
-				if (!TextUtils.isEmpty(yonaObject.getMessage()))
-				{
-					holder.txtFooterMsg.setText(yonaObject.getMessage());
-				}
-				holder.profileIconTxt.setVisibility(View.VISIBLE);
-				holder.profileIconTxt.setText(yonaObject.getNickname().substring(0, 1).toUpperCase());
-				holder.profileIconTxt.setBackground(ContextCompat.getDrawable(YonaActivity.getActivity(), R.drawable.bg_small_admin_round));
-			}
-			else if (!TextUtils.isEmpty(yonaObject.getNickname()))
-			{
-				holder.txtFooterMsg.setText(yonaObject.getNickname());
-				if (yonaObject.getNotificationMessageEnum() == NotificationMessageEnum.GOALCONFLICTMESSAGE_ANNOUNCED)
+			currentMessageItemHolder.img_status.setImageResource(currentYonaMessage.getNotificationMessageEnum().getImageId());
+		}
+	}
+
+	private String getMessageListItemTitle(NotificationEnum yonaNotificationEnum)
+	{
+		String messagetitle = "";
+		switch (yonaNotificationEnum)
+		{
+			case BUDDYCONNECTREQUESTMESSAGE:
+				messagetitle = this.activity.getString(R.string.buddyconnectrequested);
+				break;
+			case BUDDYCONNECTRESPONSEMESSAGE:
+				if (currentYonaMessage.getNotificationMessageEnum().getStatusEnum() == ACCEPTED)
 				{
-					holder.img_avtar.setImageResource(R.drawable.adult_sad);
-					holder.img_avtar.setVisibility(View.VISIBLE);
-					holder.profileIconTxt.setVisibility(View.GONE);
+					messagetitle = this.activity.getString(R.string.buddyconnectresponseaccepted);
 				}
-				else
+				else if (currentYonaMessage.getNotificationMessageEnum().getStatusEnum() == REJECTED)
 				{
-					holder.img_avtar.setVisibility(View.GONE);
-					holder.profileIconTxt.setVisibility(View.VISIBLE);
-					holder.profileIconTxt.setText(yonaObject.getNickname().substring(0, 1).toUpperCase());
-					holder.profileIconTxt.setBackground(ContextCompat.getDrawable(YonaActivity.getActivity(), R.drawable.bg_small_self_round));
+					messagetitle = this.activity.getString(R.string.buddyconnectresponserejected);
 				}
-			}
-			if (yonaObject.getLinks() != null && yonaObject.getLinks().getMarkRead() != null && !TextUtils.isEmpty(yonaObject.getLinks().getMarkRead().getHref()))
-			{
-				holder.messageContainer.setBackground(ContextCompat.getDrawable(activity, R.drawable.item_selected_gradient));
-			}
-			else
-			{
-				holder.messageContainer.setBackground(ContextCompat.getDrawable(activity, R.drawable.item_gradient));
-			}
-			holder.deleteMsg.setTag(yonaObject);
-			holder.messageContainer.setTag(yonaObject);
+				break;
+			default:
+				break;
 		}
+		return messagetitle;
+	}
+
+	private void setUpMessageItemHolderSwipe()
+	{
+		if (currentYonaMessage.getLinks() != null && currentYonaMessage.getLinks().getEdit() != null)
```

Comment:
You can replace the entire method body with this:
```
boolean isEditable = currentYonaMessage.getLinks() != null && currentYonaMessage.getLinks().getEdit() != null;
currentMessageItemHolder.swipeLayout.setRightSwipeEnabled(isEditable);
```

Reply:
Done.